### PR TITLE
BEST PRACTICE CODE REVIEW

### PR DESCRIPTION
**Pull request by Tamunokuro**
- EDIT BEST PRACTICE BRANCH WITH TODO LIST FILES

### CODE REVIEW

File: src/modules/complete.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    itemCheck.addEventListener('change', (e) => {
+      const todoList = JSON.parse(localStorage.getItem('todoList') || '[]');
+
+      todoList.forEach((item) => {
+        if ((item.completed === false) && (+e.target.id === item.index)) {
+          item.completed = true;
+          todoItems.forEach((todoI) => {
+            if (todoI.id === item.index.toString() && item.completed) { todoI.classList.add('checked'); }
+          });
+          localStorage.setItem('todoList', JSON.stringify(todoList));
```
Comment (Mujeeb4582) [inline src/modules/complete.js:15]: You can make a function for this code line such as `storedata()` and call it when required.

File: src/modules/complete.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        } else if ((item.completed === true) && (+e.target.id === item.index)) {
+          item.completed = false;
+          todoItems.forEach((todoI) => {
+            if (todoI.id === item.index.toString() && !item.completed) { todoI.classList.remove('checked'); }
+          });
+          localStorage.setItem('todoList', JSON.stringify(todoList));
```
Comment (Mujeeb4582) [inline src/modules/complete.js:17]: You can make a function for this code line such as storedata() and call it when required.

File: src/modules/todo.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    this.checked = false;
+  }
+
+  static addTodo() {
+    const todoform = document.getElementById('todo-form');
+    const todoinput = document.getElementById('todo-input');
+    let todoList = [];
+    todoform.addEventListener('submit', (e) => {
+      e.preventDefault();
+      todoList = JSON.parse(localStorage.getItem('todoList') || '[]');
```
Comment (Mujeeb4582) [inline src/modules/todo.js:26]: You can make a function for this code line such as getdata() and call it when required.

File: src/modules/todo.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    });
+  }
+
+  static deleteTodo() {
+    const del = document.querySelectorAll('.remove-icon');
+    const todoContainer = document.getElementById('todo-list');
+    del.forEach((delBtn, index) => {
+      delBtn.addEventListener('click', () => {
+        const todoList = JSON.parse(localStorage.getItem('todoList') || '[]');
+        todoList.splice(index, 1);
```
Comment (Mujeeb4582) [inline src/modules/todo.js:83]: You can make a function for this code line such as getdata() and call it when required.

File: src/modules/todo.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        todoContainer.remove();
+        todoList.forEach((todo, i) => {
+          todo.index = i + 1;
+        });
+        localStorage.setItem('todoList', JSON.stringify(todoList));
```
Comment (Mujeeb4582) [inline src/modules/todo.js:88]: You can make a function for this code line such as storedata() and call it when required.

**Review by Mujeeb4582 — CHANGES_REQUESTED** [review]
Hi @Tamunokuro ,

Good job so far!
Your design looks amazing, but there are some issues that you still need to work on to go to the next project but you are almost there!

try again

## Highlights 
- Great design✔️
- No linter errors✔️
- Good commit messages✔️

## Required Changes ♻️
Check the comments under the review.

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

You can also consider:
- Nothing to mention.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @Mujeeb4582  in your question so I can receive the notification.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your @Mujeeb4582  so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

**Comment by Tamunokuro** [comment]
Hi @Mujeeb4582 
Thank you very much for your feedback.
These are static functions which are utility functions for the class. Breaking them into parts would break the codes

**Review by Mujeeb4582 — APPROVED** [review]
## Hi @Tamunokuro 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

## Highlights

- Professional README 👍.
- Great PR description 💯.
- Functionalities implemented properly 🥇.

**Pull request merged**